src/config.rs

```diff
@@ -39,33 +51,194 @@ pub struct Config {
             .into_iter()
             .map(|rule| InstantiatedRule::new(rule.clone(), self))
-            .filter(|rule| match self.rule.as_ref() {
-                Some(rule_arg) => &rule.meta.name == rule_arg,
-                None => true,
-            })
             .collect::<Vec<_>>();
         if instantiated_rules.is_empty() {
             panic!("Invalid rule name: {:?}", self.rule.as_ref().unwrap());
         }
         instantiated_rules
     }
 }
+
+impl ConfigBuilder {
+    pub fn default_rule_configurations(&mut self) -> &mut Self {
+        self.rule_configurations = Some(
+            self.all_rules
+                .as_ref()
+                .expect("must call .all_rules() before calling .default_rule_configurations()")
+                .into_iter()
+                .map(RuleConfiguration::default_for_rule)
+                .collect(),
+        );
+        self
+    }
+}
+
+#[derive(Clone)]
+pub struct ParsedConfigFile {
+    pub path: PathBuf,
+    pub content: ParsedConfigFileContent,
+}
+
+#[derive(Clone, Deserialize)]
+pub struct ParsedConfigFileContent {
+    #[serde(rename = "rules")]
+    rules_by_name: HashMap<String, RuleConfigurationValue>,
+}
+
+impl ParsedConfigFileContent {
+    pub fn rules(&self) -> impl Iterator<Item = RuleConfiguration> + '_ {
+        self.rules_by_name
+            .iter()
+            .map(|(rule_name, rule_config)| rule_config.to_rule_configuration(rule_name))
+    }
+}
+
+#[derive(Copy, Clone, Debug, PartialEq, Eq, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub enum ErrorLevel {
+    Error,
+    Off,
+}
+
+#[derive(Clone, Deserialize)]
+pub struct RuleConfigurationValue {
+    pub level: ErrorLevel,
+}
+
+impl RuleConfigurationValue {
+    pub fn to_rule_configuration(&self, rule_name: impl Into<String>) -> RuleConfiguration {
+        let rule_name = rule_name.into();
+        RuleConfiguration {
+            name: rule_name,
+            level: self.level,
+        }
+    }
+}
+
+#[derive(Clone)]
+pub struct RuleConfiguration {
+    pub name: String,
+    pub level: ErrorLevel,
+}
+
+impl RuleConfiguration {
+    pub fn default_for_rule(rule: &Arc<dyn Rule>) -> Self {
+        Self {
+            name: rule.meta().name,
+            level: ErrorLevel::Error,
+        }
+    }
+}
+
+fn load_config_file() -> ParsedConfigFile {
+    let config_file_path = find_config_file();
+    let config_file_contents =
+        fs::read_to_string(&config_file_path).expect("Couldn't read config file contents");
+    let parsed = serde_yaml::from_str(&config_file_contents).expect("Couldn't parse config file");
+
+    ParsedConfigFile {
+        path: config_file_path,
+        content: parsed,
+    }
+}
+
+const CONFIG_FILENAME: &str = ".tree-sitter-lint.yml";
+
+fn find_config_file() -> PathBuf {
+    find_filename_in_ancestor_directory(
+        CONFIG_FILENAME,
+        env::current_dir().expect("Couldn't get current directory"),
+    )
+    .expect("Couldn't find config file")
```

Comment:
I'm tending to think that a config file is _required_ when using from the command-line

Because if we (a) don't bake in a set of "default" rules a la ESLint (which my instinct is no we wouldn't do that) and (b) there's no way to "fully specify" a rule via the command line (it has to be compiled Rust code) then there's no way to "have any known compiled rules" without a config file to instruct us?

And then that would have the nice property of being able to create the `.tree-sitter-lint/` directory (where the "custom local binary" project would be generated) in the same directory where we found the `.tree-sitter-lint.yml` config file (maybe that's configurable if you don't want to eg have to avoid committing `.tree-sitter-lint/` and it could use some "shared" system directory instead)?

So that thinking is somewhat influenced by the direction of doing the "custom local binary", not sure where I'd land wrt the same question if we were doing eg dynamic-linking